Sources/SocketForwarder/LRUCache.swift

```diff
@@ -0,0 +1,107 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2025 Apple Inc. and the container project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+
+struct KeyExistsError: Error {}
+
+class LRUCache<K: Hashable, V> {
+    private class Node {
+        fileprivate var prev: Node?
+        fileprivate var next: Node?
+        fileprivate let key: K
+        fileprivate let value: V
+
+        init(key: K, value: V) {
+            self.prev = nil
+            self.next = nil
+            self.key = key
+            self.value = value
+        }
+    }
+
+    private var head: Node?
+    private var tail: Node?
+    private var members: [K: Node]
+
+    public init() {
```

Comment:
nit: The constructor and methods likely don't need to be public as the type itself isn't